xds/internal/client/lds_test.go

```diff
@@ -933,8 +990,428 @@ func (s) TestUnmarshalListener_ServerSide(t *testing.T) {
 			if err != nil && !strings.Contains(err.Error(), test.wantErr) {
 				t.Fatalf("UnmarshalListener(%v) = %v wantErr: %q", test.resources, err, test.wantErr)
 			}
-			if !cmp.Equal(gotUpdate, test.wantUpdate, cmpopts.EquateEmpty()) {
-				t.Errorf("UnmarshalListener(%v) = %v want %v", test.resources, gotUpdate, test.wantUpdate)
+			if diff := cmp.Diff(test.wantUpdate, gotUpdate, cmpopts.EquateEmpty()); diff != "" {
+				t.Errorf("UnmarshalListener(%+v) returned unexpected diff (-want, +got):\n%s", test.resources, diff)
+			}
+		})
+	}
+}
+
+func (s) TestGetFilterChain(t *testing.T) {
+	tests := []struct {
+		desc             string
+		inputFilterChain *v3listenerpb.FilterChain
+		wantFilterChain  *FilterChain
+		wantErr          bool
+	}{
+		{
+			desc:             "empty",
+			inputFilterChain: nil,
+		},
+		{
+			desc: "unsupported destination port",
```

Comment:
Why don't we want an error for this & other things that look like they would fail?